src/sentry/incidents/logic.py

```diff
@@ -1452,3 +1453,47 @@ def translate_aggregate_field(aggregate, reverse=False):
                 if translated_field == column:
                     return aggregate.replace(column, field)
     return aggregate
+
+
+def alert_rule_data_requires_async_lookup(organization, user, data, write_input_channel_id=False):
+    try:
+        from sentry.incidents.endpoints.serializers import AlertRuleTriggerActionSerializer
+
+        for trigger in data["triggers"]:
+            for action in trigger["actions"]:
+                a_s = AlertRuleTriggerActionSerializer(
+                    context={
+                        "organization": organization,
+                        "access": SystemAccess(),
+                        "user": user,
+                        "use_async_lookup": True,
+                    },
+                    data=action,
+                )
+                if a_s.is_valid():
+                    if a_s.validated_data["type"].value in AlertRuleTriggerAction.INTEGRATION_TYPES:
```

Comment:
@iProgramStuff Can we just check `a_s.validated_data["type"].value == AlertRuleTriggerAction.Type.SLACK.value` here instead of having two checks on `a_s.validated_data["type"].value `?

Reply:
Yes, that's my bad.

Reply:
Fixed.